From 8ebe7201ae5b8b56f5b9ad67df73318b24fae457 Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Mon, 11 Jun 2018 23:27:38 -0700
Subject: [PATCH 1682/2367] ASoC: SOF: fix warning about assigning __le to u16
 type

This patchs addresses the warning about assigning a __le value
to a u16 type.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/topology.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index cdf21b0..dfaae19 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -290,7 +290,7 @@ static int get_token_u16(void *elem, void *object, u32 offset, u32 size)
 	struct snd_soc_tplg_vendor_value_elem *velem = elem;
 	u16 *val = object + offset;
 
-	*val = velem->value;
+	*val = (u16)le32_to_cpu(velem->value);
 	return 0;
 }
 
-- 
2.7.4

